src/meepgeom.cpp

```diff
@@ -2999,12 +2999,23 @@ void material_grids_addgradient(double *v, size_t ng, size_t nf,
   for (int i = 0; i < 3; i++) {
     for (int ii = 0; ii < adjoint_dft_chunks[i].size(); ii++) {
       delete adjoint_dft_chunks[i][ii];
-      delete forward_dft_chunks[i][ii];
     }
   }
 
 } // material_grids_addgradient
 
+void delete_forward_dft_chunk(std::vector<meep::dft_fields *> fields_f) {
```

Comment:
It seems like it would make more sense to have a `dft_fields::delete_dft_chunks` method, and then just call it for each element of the array.